### PR TITLE
[Wait for #1829] [lstm] implement bidirectional lstm forward

### PR DESCRIPTION
**Pull request by lhs8928**
[lstm] implement bidirectional lstm forward
    
     - For now only support forward bidirectional lstm
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

### CODE REVIEW

File: nntrainer/layers/lstm.cpp [hunk 1]
```diff
@@ -31,16 +31,84 @@ enum LSTMParams {
...
+  reverse_bias_h,
+  reverse_bias_ih,
+  reverse_bias_hh,
+  reverse_hidden_state,
+  reverse_cell_state,
+  reverse_ifgo,
   dropout_mask
 };
 
+void batch_first_forwarding(
```
Comment (kparichay) [inline nntrainer/layers/lstm.cpp:45]: make this function static, and doxygen for it.

File: nntrainer/layers/lstm.cpp [hunk 2]
```diff
@@ -31,16 +31,84 @@ enum LSTMParams {
...
+
+  for (unsigned int batch = 0; batch < batch_size; ++batch) {
+    const Tensor input_sample = input_.getBatchSlice(batch, 1);
+    Tensor hidden_state_sample = hidden_state_.getBatchSlice(batch, 1);
+    Tensor cell_state_sample = cell_state_.getBatchSlice(batch, 1);
+    Tensor ifgo_sample = ifgo_.getBatchSlice(batch, 1);
+
+    for (unsigned int t = 0; t < max_timestep; ++t) {
+      Tensor input = input_sample.getSharedDataTensor(
+        {feature_size}, (reverse ? max_timestep - 1 - t : t) * feature_size);
```
Comment (kparichay) [inline nntrainer/layers/lstm.cpp:65]: :100:

File: nntrainer/layers/lstm.cpp (hunk 1 above)
Comment (lhs8928) [reply nntrainer/layers/lstm.cpp]: Done

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Review by kparichay — APPROVED** [review]
LGTM

**Pull request merged**